junit-platform-launcher/src/main/java9/org/junit/platform/launcher/listeners/FlightRecordingListener.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2015-2020 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * https://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junit.platform.launcher.listeners;
+
+import static org.apiguardian.api.API.Status.EXPERIMENTAL;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.stream.Collectors;
+
+import jdk.jfr.Category;
+import jdk.jfr.Description;
+import jdk.jfr.Event;
+import jdk.jfr.Label;
+
+import org.apiguardian.api.API;
+import org.junit.platform.engine.TestExecutionResult;
+import org.junit.platform.launcher.TestExecutionListener;
+import org.junit.platform.launcher.TestIdentifier;
+import org.junit.platform.launcher.TestPlan;
+
+/**
+ * A {@link TestExecutionListener} that generates Java Flight Recorder
+ * events.
+ *
+ * @see <a href="https://openjdk.java.net/jeps/328">JEP 328: Flight Recorder</a>
+ * @since 1.7
+ */
+@API(status = EXPERIMENTAL, since = "1.7")
+public class FlightRecordingListener implements TestExecutionListener {
+
+	private final AtomicReference<TestPlanExecutionEvent> testPlanExecutionEvent;
+	private final Map<String, TestExecutionEvent> testExecutionEventMap;
+
+	public FlightRecordingListener() {
+		this.testPlanExecutionEvent = new AtomicReference<>();
+		this.testExecutionEventMap = new ConcurrentHashMap<>();
+	}
+
+	@Override
+	public void testPlanExecutionStarted(TestPlan plan) {
+		TestPlanExecutionEvent event = new TestPlanExecutionEvent();
+		event.containsTests = plan.containsTests();
+		event.roots = plan.getRoots().stream().map(TestIdentifier::getUniqueId).collect(Collectors.joining("\t"));
```

Comment:
If this is a path, consider using "/" as a separator, i.e. "id1 / id2 / id3". Tabs may result in different number of spaces, or may not work at all, if visualized in a GUI.

Reply:
The format defined by `UniqueId` uses `/` as an element separator char -- so, that's already taken. See details at:

https://github.com/junit-team/junit5/blob/19785134f95ba100346eb2d8b0c24f707535ebe4/junit-platform-engine/src/main/java/org/junit/platform/engine/UniqueIdFormat.java#L42

Using tab `\t` is not a good choice either. Guess, we need to define a human-visible, URI-encodeable unique-id separator char.

Reply:
Using `#` for the time being.